ReactNativeClient/lib/components/screens/note.js

```diff
@@ -604,6 +625,101 @@ class NoteScreenComponent extends BaseScreenComponent {
 		this.setState({ noteTagDialogShown: true });
 	}
 
+	recordAudio_onPress() {
+		if (!this.state.note || !this.state.note.id) return;
+
+		if (this.state.isRecording) {
+			this.setState({ isRecording: false });
+			this.onStopRecord();
+		} else {
+			this.setState({ isRecording: true });
+			this.onStartRecord();
+		}
+	}
+
+	async onStartRecord() {
+		if (Platform.OS === 'android') {
+			try {
+				const granted = await PermissionsAndroid.request(
+					PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+					{
+				  		title: 'Permissions for write access',
+				  		message: 'Give permission to your storage to write a file',
+				  		buttonPositive: 'ok',
+					},
+			  	);
+				if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+					console.log('You can use the storage');
+				} else {
+					console.log('permission denied');
+					return;
+				}
+			} catch (err) {
+				console.warn(err);
+				return;
+			}
+		  }
+		  if (Platform.OS === 'android') {
+			try {
+				const granted = await PermissionsAndroid.request(
+					PermissionsAndroid.PERMISSIONS.RECORD_AUDIO,
+					{
+						title: 'Permissions for write access',
+						message: 'Give permission to your storage to write a file',
+						buttonPositive: 'ok',
+					},
+				);
+				if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+					console.log('You can use the camera');
+				} else {
+					console.log('permission denied');
+					return;
+				}
+			} catch (err) {
+				console.warn(err);
+				return;
+			}
```

Comment:
The above seems to repeat the logic just above it or am I missing something?

Reply:
It is checking for 2 different permissions. It needs WRITE_EXTERNAL_STORAGE and RECORD_AUDIO. I need to update the messages though.